src/components/Dialogs/Conditions/OrgUnitCondition.js

```diff
@@ -72,6 +79,111 @@ const OrgUnitCondition = ({
         />
     )
 }
+const getMetaDataId = (id) =>
+    ouIdHelper.hasLevelPrefix(id) || ouIdHelper.hasGroupPrefix(id)
+        ? ouIdHelper.removePrefix(id)
+        : id
+
+// VERSION-TOGGLE: rename to OrgUnitCondition when 42 is lowest supported version
+const OrgUnitConditionMinVersionV42 = ({
+    addMetadata,
+    addParentGraphMap,
+    condition,
+    onChange,
+    metadata,
+    parentGraphMap,
+}) => {
+    const { rootOrgUnits, currentUser } = useCachedDataQuery()
+    const selected = useMemo(() => {
+        const idsString = condition.split(':')?.[1] ?? ''
+
+        if (idsString.length === 0) {
+            return []
+        }
+
+        return idsString.split(';').map((id) => {
+            const isOuTreeItem =
+                !ouIdHelper.hasLevelPrefix(id) &&
+                !ouIdHelper.hasGroupPrefix(id) &&
+                !id.startsWith('USER_ORGUNIT')
+            const metaDataId = getMetaDataId(id)
+            const item = {
+                id,
+                // We store IDs for levels and groups in the metadata store without prefix
+                name:
+                    metadata[metaDataId].displayName ||
+                    metadata[metaDataId].name,
+            }
+
+            if (isOuTreeItem) {
+                item.path = getOuPath(id, metadata, parentGraphMap)
+            }
+
+            return item
+        })
+    }, [condition, metadata, parentGraphMap])
+    const onSelect = useCallback(
+        ({ items }) => {
+            if (items.length === 0) {
+                onChange('')
+            } else {
+                const { forMetadata, forParentGraphMap, itemIds } =
+                    items.reduce(
+                        (acc, item) => {
+                            const metaDataId = getMetaDataId(item.id)
+                            acc.itemIds.push(item.id)
+                            acc.forMetadata[metaDataId] = {
+                                id: metaDataId,
+                                name: item.name || item.displayName,
+                                displayName: item.displayName,
```

Comment:
is there any need for shortName support?

Reply:
Perhaps... But I'd classify it as out-of-scope for this PR because:
1. This isn't really new code, it's just been copied from the old component, and we weren't taking `shortName` into account before
2. If we tackle this here, we should really scan the full app for places where we write to the `metaData` store and ensure we take the `shortName` into account everywhere

Shall we create a JIRA ticket for this?

Reply:
I'll check into what Jira tickets we have already for shortName.